processor/src/advice/mod.rs

```diff
@@ -88,21 +91,8 @@ pub trait AdviceProvider {
     /// Returns an error if the advice tape does not contain two words.
     fn read_tape_dw(&mut self) -> Result<[Word; 2], ExecutionError>;
 
-    /// Writes the provided value at the head of the advice tape.
-    fn write_tape(&mut self, value: Felt);
-
-    /// Fetch a keyed tape from the values map, reversing and appending it to the advice tape.
-    ///
-    /// Note: this operation shouldn't consume the map element so it can be called multiple times
-    /// for the same key.
-    ///
-    /// # Example
-    /// Given an advice map `[a,b,c]`, and a map `x |-> [d,e,f]`, a call `write_tape_from_map(x)`
-    /// will result in `[a,b,c,f,e,d]` for the advice tape, and will preserve `x |-> [d,e,f]`.
-    ///
-    /// # Errors
-    /// Returns an error if the key was not found in a key-value map.
-    fn write_tape_from_map(&mut self, key: Word) -> Result<(), ExecutionError>;
+    /// Writes values to the end of the tape, consuming the provided source.
```

Comment:
question: Is this correct? Does it consume the provided source?

Reply:
Yes, it will be consumed as we take `AdviceSource` as owned. Since it implements `Copy`, shouldn't be a major issue. I was thinking of taking this as ref as it might either be a `Felt` that will have to be copied, regardless, or a key to a map, that will also be queried by reference.

The main reason to keep as owned is if we need to put some reference to a big type there, we might as well just use a box/rc/pointer, as long as we don't leak any lifetimes.

Example:

```rust
pub enum AdviceSource {
  Value(Felt),
  Map(Word),
  Foo(Rc<Bar>),
}```

Reply:
I think the confusion here may be around comment wording. The parameter is consumed, but the source is not. Maybe we should change the comment to something like:

Writes values specified by the source to the head of the advice tape.

Reply:
That indeed seems to be more comprehensible; updated!